docs/arcadia-users-group/20240305-github-templates/lesson.md

```diff
@@ -0,0 +1,88 @@
+- [.github/PULL_REQUEST_TEMPLATE.md](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.github/PULL_REQUEST_TEMPLATE.md): The .github folder controls some behavior of the GitHub repository itself. The PULL_REQUEST_TEMPLATE.md is a text document where users can provide the text that they want to appear by default each time a new PR is opened. Often, these files remind users of the list of things that need to be done before a PR is read to be reviewed. Below we show this in action. On the left is the typical screen a user is prompted with when they open a pull request in a repository without a PR template. On the right is the PR template as well as that template rendered in the pull request.
+![](pr-template.png)
+
+### Checking for accuracy, enforcing conventions, and making your life easier
+
+- [.gitignore](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.gitignore): The `.gitignore` file lists files and directories that should be ignored by git, meaning they won't be tracked or included in the repository. This typically includes temporary files, local configuration files, and sensitive information. Using a .gitignore file helps keep the repository clean and prevents the accidental inclusion of irrelevant or sensitive files.
+- [.github/lint.yml](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.github/lint.yml): The `.github/lint.yml` file is a GitHub Actions configuration file that sets up linting for the repository. Linting automatically checks the code for stylistic errors, programming errors, and other potential issues according to predefined rules. This helps maintain code quality and consistency across the project, making it easier to read, maintain, and contribute to (see [this AUG lesson](https://training.arcadiascience.com/arcadia-users-group/20240206-intro-to-formatting-and-linting/lesson/) for more details).
+- [.pre-commit-config.yaml](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.pre-commit-config.yaml): The `.pre-commit-config.yaml` file configures pre-commit hooks for the repository. Pre-commit hooks are scripts that run automatically on each commit to check for common issues, such as syntax errors, formatting issues, or failing tests. This setup helps catch and fix problems early in the development process, improving code quality and reducing the time spent on code reviews.
+- [.vscode/extensions.json](https://github.com/Arcadia-Science/python-analysis-template/tree/main/.vscode/extensions.json): The `.vscode/extensions.json` file is a configuration file for Visual Studio Code that specifies recommended extensions for the project. This file will prompt VS Code to install the ruff linter.
+- [Makefile](https://github.com/Arcadia-Science/python-analysis-template/tree/main/Makefile): The `Makefile` helps automate code quality and style checks, making it easier for developers to run all of these checks at once.
+
+## Benefits of templates
```

Comment:
another benefit: consistency across the org (so we don't have different projects using different package managers, directory strutures, linting tools, etc)